redisson/src/main/java/org/redisson/api/MapOptions.java

```diff
@@ -89,7 +96,7 @@ public static <K, V> MapOptions<K, V> defaults() {
      * @return MapOptions instance
      */
     public MapOptions<K, V> writer(MapWriter<K, V> writer) {
-        this.writer = writer;
+        this.writer = new RetryableMapWriter<>(this, writer);
```

Comment:
don't apply RetryableMapWriter if writerRetryAttempts == 0

Reply:
One question: when create MapOptions,  `.writerRetryAttempts()` maybe called after`.writerAsync()`, like this:

```
           MapOptions.<>defaults()
                .writerAsync(new MapWriterAsync<String, String>() {
                 ...
                })
                .writerRetryAttempts(expectedRetryAttempts);
```

In this case, `RetryableMapWriter` won't be applied even if users intend it to be.